### ISSUE TITLE
Make sure `no-unnecessary-act` detects unnecessary `act` called before the `render` method

### ISSUE THREAD

**Issue by fabiendem** [testing-library/eslint-plugin-testing-library#551]
### What rule do you want to change?

no-unnecessary-act

### Does this change cause the rule to produce more or fewer warnings?

More warnings

### How will the change be implemented?

I don't have a plan to implement the change yet.
The idea is to improve the rule `no-unnecessary-act` so it can detect unnecessary `act` which are called _before_ a render.
https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-unnecessary-act.md
Those unnecessary make the codebase noisy.

### Example code

```ts
act(() => {
  someSetupMethod();
});

render(<Example />);
```

### How does the current rule affect the code?

This unnecessary `act` is not detected

### How will the new rule affect the code?

This unnecessary `act` is detected

### Do you want to submit a pull request to change the rule?

Yes, but need help

**Comment by Belco90** [comment]
Hey @fabiendem! Thanks for the suggestion, it's an interesting one. It would be great if you finally make some time and implement it yourself. I'll try to help as much as I can!

**Comment by fabiendem** [comment]
Thank you 👍

**Comment by fabiendem** [comment]
Feature is still relevant, I didn't have a chance to tackle it yet.

**Comment by Belco90** [comment]
This may be tricky to check tho. What if the developer is rendering the component in the `beforeEach` hook? This rule won't find any valid render within the tests so all acts could be potentially reported.

**Comment by fabiendem** [comment]
We have this rule in place: https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-render-in-setup.md
So this is not an issue for us. But surely not everyone will have this rule enabled.

But even, we would want to detect any `act` before a `render` call.
We can start by detecting this in the same blocks. Then detect `acts` in the `before*` blocks as well.

**Comment by Belco90** [comment]
Rules must not depend on each other, so we have to assume this scenario is possible.

> We can start by detecting this in the same blocks. Then detect acts in the before* blocks as well.

LGTM! So this rule would report acts for the current criteria OR for acts called before a render within the test block (ignoring them if no render is found).

**Comment by fabiendem** [comment]
> Rules must not depend on each other, so we have to assume this scenario is possible.

yeah definitely

> LGTM! So this rule would report acts for the current criteria OR for acts called before a render within the test block (ignoring them if no render is found).

yeah seems legit
